fatjdbc/src/main/java/com/qubole/quark/fatjdbc/executor/DDLPlanExecutor.java

```diff
@@ -98,27 +98,27 @@ public QuarkMetaResultSet execute(ParserResult result)
     }
     if (sqlNode instanceof SqlCreateQuarkDataSource) {
       int id = executeCreateDataSource((SqlCreateQuarkDataSource) sqlNode);
-      connection.resetSelectParser();
+      connection.setIsDirty();
```

Comment:
Will this class also be used by Quark Service ? Then that implies:

1. isDirty flag should be moved into `ParserFactory`.
2. DDLPlanExecutor should be using `parserFactory` instead of `connection`.